docs/DeveloperGuide.md

```diff
@@ -135,107 +135,78 @@ The `Model` component,
+**API** : [`Storage.java`](https://github.com/AY2122S1-CS2103T-T11-2/tp/blob/master/src/main/java/lingogo/storage/Storage.java)
 
 <img src="images/StorageClassDiagram.png" width="550" />
 
 The `Storage` component,
 * can save both address book data and user preference data in json format, and read them back into corresponding objects.
-* inherits from both `AddressBookStorage` and `UserPrefStorage`, which means it can be treated as either one (if only the functionality of only one is needed).
+* inherits from both `FlashcardAppStorage` and `UserPrefStorage`, which means it can be treated as either one (if only the functionality of only one is needed).
 * depends on some classes in the `Model` component (because the `Storage` component's job is to save/retrieve objects that belong to the `Model`)
 
 ### Common classes
 
-Classes used by multiple components are in the `seedu.addressbook.commons` package.
+Classes used by multiple components are in the `lingogo.commons` package.
 
 --------------------------------------------------------------------------------------------------------------------
 
 ## **Implementation**
 
 This section describes some noteworthy details on how certain features are implemented.
 
-### \[Proposed\] Undo/redo feature
+### Filter feature
 
-#### Proposed Implementation
+#### Implementation
 
-The proposed undo/redo mechanism is facilitated by `VersionedAddressBook`. It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`. Additionally, it implements the following operations:
+The filter feature is facilitated by `ModelManager`. It extends `Model` and implements `updateFilteredFlashcardList` 
+which returns an unmodifiable view of filtered flashcards in the GUI.
 
-* `VersionedAddressBook#commit()` — Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` — Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` — Restores a previously undone address book state from its history.
+The filter feature also relies on a nested `FilterBuilder` class within `FilterCommand`. Multiple filters can be 
+given by the user in one command, however only one predicate (filter) can be accepted by 
+`Model::updateFilteredFlashcList` to produce the filtered flashcards. `FilterBuilder` helps by generating multiple 
+predicates from the user input, and then combining them into a single predicate. 
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+`FilterBuilder` is also a mutable class which allows processed user inputs to be directly set as variables within a 
+`FilterBuilder` instance. A mutable design is acceptable for `FilterBuilder` since it acts as a throwaway variable, 
+with only a one-time usage within `FilterCommand`. Furthermore, setVariable methods in `FilterBuilder` reduces the 
```

Comment:
```suggestion
`FilterBuilder` instance. A mutable design is acceptable for `FilterBuilder` since it only has a one-time usage within `FilterCommand`. Furthermore, setVariable methods in `FilterBuilder` reduces the 
```